### PR TITLE
Initial docs

### PR DESCRIPTION
**Pull request by lunarserge**
Docs: high-level overview & example.
Built documentation is here https://lunarserge.github.io/facere-sensum, but looks ugly. Filed #17 to fix.

### CODE REVIEW

File: doc/intro.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
+############
+Introduction
+############
+
+``facere-sensum`` is a general-purpose metrics framework that helps put a meaningful number on anything. It allows to combine individual metrics into a higher-level metric that represents collective behavior as a single indicator.
```
Comment (dbabokin) [inline doc/intro.rst:5]: "helps put"? what does this mean?

File: doc/intro.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Here we have several levels of metrics:
+
+* On the lowest level there are individual metrics for how well specific project landing page is found using specific search term.
+* Next level combines search terms for a particular project and yields a single-number metric for how well project landing page is found across *all* the search terms.
+* Finally, the top level - top of the tree - is just one single number reflecting how well the company does SEO across *all* its projects.
+
+This example is using homogeneous metrics, but in practice metrics can get heterogeneous. E.g., you can imagine even higher level which combines SEO efficiency with something different like uptime metric for landing pages. To abstract away such differences, ``facere-sensum`` is using normalized metric values at all levels. The values must be a number between ``0`` and ``1``, with ``0`` indicating inferior performance, ``1`` indicating perfect performance and other values indicating varying level of performance in between. This is how it works at various levels:
```
Comment (dbabokin) [inline doc/intro.rst:23]: "a number between 0 and 1" => "a real number between 0 and 1"?

File: doc/intro.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* Lowest level metrics should be defined to support this directly. E.g., in the SEO example, lowest level metric (for project / search term pair) can be defined as ``1`` if the landing page shows up first in search results, 0.9 if it shows up second and so forth. The metric becomes ``0`` if the landing page is not found in the top ten search results.
+* Higher level metrics are calculated automatically by ``facere-sensum`` as a weighted sum of the corresponding lower-level metric values, with weights totaling to ``1``. This happens on all levels above the lowest. Since lower-level metric values are between ``0`` and ``1`` and weights are totaling to ``1`` this yields higher level metric values between ``0`` and ``1`` as well:
```
Comment (dbabokin) [inline doc/intro.rst:26]: "weighted sum" or "weighted average"? The first one seems to be incorrect, but I'm not sure about the second.

**Review by dbabokin — APPROVED** [review]
Looks good, but I would suggest to add a pipeline to verify that rsts are not malformed. I'm not sure if Sphinx has this functionality. You can try running it in command line and see if there are errors. Just for reference, this [pipeline](https://github.com/ispc/ispc/blob/main/.github/workflows/docs.yml) does this in ISPC. It runs [build script](https://github.com/ispc/ispc/blob/main/docs/build.sh) and checks if it fails.

File: doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+   intro
+   installation
+   examples
+
+******************
+Indices and tables
+******************
+
+* :ref:`genindex`
+* :ref:`modindex`
```
[the remark this answers is not in the document]
Comment (lunarserge) [reply doc/index.rst:18]: Decided to remove 'Indices and tables' section as I don't have the content for it

File: doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+   intro
+   installation
+   examples
+
+******************
+Indices and tables
+******************
+
+* :ref:`genindex`
```
[the remark this answers is not in the document]
Comment (lunarserge) [reply doc/index.rst:17]: Decided to remove 'Indices and tables' section as I don't have the content for it

File: doc/intro.rst (hunk 1 above)
Comment (lunarserge) [reply doc/intro.rst:5]: Will address in an upcoming PR

File: doc/intro.rst (hunk 2 above)
Comment (lunarserge) [reply doc/intro.rst:23]: yes, but a 'floating-point number' would be more correct it seems

File: doc/intro.rst (hunk 3 above)
Comment (lunarserge) [reply doc/intro.rst:26]: right - 'weighted average'

**Pull request merged**